### PR TITLE
client: Make beacon sync more robust

### PR DESCRIPTION
**Pull request by g11tech**
Handle few recovery and reorg scenarios to make beacon sync more robust
plus a small refac for better code clarity

@holgerd77 we can target this for `v5 maintenance` if we want, although we can also skip it as beacon sync is relatively new.

### CODE REVIEW

**Comment by holgerd77** [comment]
Ugh, sorry. This got a bit lost.

@g11tech What is the state of this PR here? Does this only need a conflict resolution and branch update? Or did you mean this PR when mentioning that you still have got some changes locally?

**Comment by g11tech** [comment]
> Ugh, sorry. This got a bit lost.
> 
> @g11tech What is the state of this PR here? Does this only need a conflict resolution and branch update? Or did you mean this PR when mentioning that you still have got some changes locally?

yes, there are one set of changes we can add in, other set i still have to figure out and fix w.r.t. a skeleton subchain merge.
I will resolve conflict and update! :slightly_smiling_face:

File: packages/client/lib/sync/skeleton.ts
Comment (acolytec3) [inline packages/client/lib/sync/skeleton.ts:365]:
```suggestion
        throw Error(`Blocks don't extend canonical subchain`)
```

**Review by acolytec3 — COMMENTED** [review]
Generally looks good to me (one nit).  Once I sync sepolia with it, will approve.

File: packages/client/lib/sync/sync.ts
Comment (acolytec3) [inline packages/client/lib/sync/sync.ts:178]:
```suggestion
          `Finishing up sync with the current fetcher ${heightStr}
```
There's a weird newline in this log (probably from `prettier`).  I can't add it in this comment but you need to delete the newline as well that runs into line 179

Comment (g11tech) [reply packages/client/lib/sync/sync.ts]: done!

**Review by acolytec3 — APPROVED** [review]
LGTM with the fixes on the logging

**Comment by acolytec3** [comment]
Updated master in UI after checking locally.  Will merge once the tests run

**Pull request merged**
